Description: lock folder
 Put firehol lock files into a specific lock folder.
Origin: debian
Author: Jerome Benoit <calculus@rezozer.net>
Last-Update: 2014-08-06

--- a/sbin/firehol.in
+++ b/sbin/firehol.in
@@ -420,8 +420,13 @@
 FIREHOL_TMP="${FIREHOL_DIR}/firehol-tmp.sh"
 
 # Redhat like status info for startup script
-FIREHOL_LOCK_DIR="/var/lock/subsys"
-test ! -d "${FIREHOL_LOCK_DIR}" && FIREHOL_LOCK_DIR="/var/lock"
+if [ -d "/var/lock/subsys" ]
+then
+	FIREHOL_LOCK_DIR="/var/lock/subsys/firehol"
+else
+	FIREHOL_LOCK_DIR="/var/lock/firehol"
+fi
+
 
 if [ -d "/var/spool" ]
 then
@@ -596,6 +601,14 @@
 EOFMTL
 
 
+# Make sure we have a directory for our locks.
+if [ ! -d "${FIREHOL_LOCK_DIR}" ]
+then
+	"${MKDIR_CMD}" "${FIREHOL_LOCK_DIR}"		|| exit 1
+	"${CHOWN_CMD}" root:root "${FIREHOL_LOCK_DIR}"	|| exit 1
+	"${CHMOD_CMD}" 700 "${FIREHOL_LOCK_DIR}"	|| exit 1
+fi
+
 # Make sure we have a directory for our data.
 if [ ! -d "${FIREHOL_SPOOL_DIR}" ]
 then
